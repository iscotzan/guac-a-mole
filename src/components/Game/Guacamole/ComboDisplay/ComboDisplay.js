import * as React from "react";
import { Circle } from "rc-progress";

const ComboDisplay = props => {
  const { scoreMultiplier, comboMaxLevel } = props;
  let comboPercentLevel = (scoreMultiplier / comboMaxLevel) * 100;
  return (
    <div className="comboDisplayContainer">
      <div className={"comboDisplayText"}>x{scoreMultiplier}</div>
      <Circle
        percent={comboPercentLevel}
        strokeWidth="6"
        strokeLinecap="round"
        className={"comboDisplayCircle"}
        strokeColor={"#91ff3f"}
      />
    </div>
  );
};

export default ComboDisplay;
