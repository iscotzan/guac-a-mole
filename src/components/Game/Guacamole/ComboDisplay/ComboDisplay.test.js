import React from "react";
import ComboDisplay from "./ComboDisplay";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<ComboDisplay scoreMultiplier={1} comboMaxLevel={6} />);
describe("<ComboDisplay />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
