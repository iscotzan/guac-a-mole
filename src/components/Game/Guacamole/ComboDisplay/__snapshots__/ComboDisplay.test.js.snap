// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComboDisplay /> renders without crashing 1`] = `
<div
  className="comboDisplayContainer"
>
  <div
    className="comboDisplayText"
  >
    x
    1
  </div>
  <Progress
    className="comboDisplayCircle"
    gapPosition="top"
    percent={16.666666666666664}
    prefixCls="rc-progress"
    strokeColor="#91ff3f"
    strokeLinecap="round"
    strokeWidth="6"
    style={Object {}}
    trailColor="#D9D9D9"
    trailWidth={1}
  />
</div>
`;
